### PR TITLE
Handle more than one Python module.

### PR DESCRIPTION
**Pull request by bzed**
Github issue #227.

Installing module directories is not handled yet,
only single file modules.

Also fix the RedHat installation of collectd-python for RedHat
Github issue #239.

### CODE REVIEW

File: manifests/plugin/python.pp [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
-  $order = '10',
+  $interval = undef,
 ) {
-  include collectd::params
+  require collectd::params
 
-  validate_hash($config)
+  if (
+      $::osfamily == 'Redhat' and
+      operatingsystemmajrelease < 7
```
Comment (txaj) [inline manifests/plugin/python.pp]: does this even gets parsed ?!

File: manifests/plugin/python.pp [hunk 2]
```diff
@@ -20,26 +22,39 @@
...
-      require => File["${name}.load"],
-      notify  => Service['collectd'],
+  $ensure_modulepath = $ensure ? {
+      'absent' => $ensure,
+      default  => 'directory',
+  }
+
+  file { $modulepath :
+      ensure  => $ensure_modulepath,
+      mode    => '0755',
```
Comment (txaj) [inline manifests/plugin/python.pp]: is o+rx necessary ?

**Comment by txaj** [comment]
Could you write some tests as well ? thanks

File: manifests/plugin/python.pp (hunk 1 above)
Comment (bzed) [inline manifests/plugin/python.pp]: heh, ouch, seems I missed a $::, but yes, it parses well for some reason :)

File: manifests/plugin/python.pp (hunk 2 above)
Comment (bzed) [inline manifests/plugin/python.pp]: Probably not, but it should not contain sensitive stuff, there are config files for that.

Comment (txaj) [inline manifests/plugin/python.pp]: I'm not a python guy but the least privilege principle should be applied here

**Comment by bzed** [comment]
I'll do if I find the time to do so, unfortunately thats the thing I'm missing most of the time :(

File: manifests/plugin/python.pp (hunk 2 above)
Comment (jesusaurus) [inline manifests/plugin/python.pp]: Agreed, we shouldn't change the permissions without need.

File: manifests/plugin/python.pp [hunk 3]
```diff
@@ -20,26 +22,39 @@
...
+      owner   => root,
+      group   => root,
```
Comment (txaj) [inline manifests/plugin/python.pp]: should be $collectd::params::root_group

File: manifests/plugin/python.pp (hunk 2 above)
Comment (bzed) [inline manifests/plugin/python.pp]: the directory is being created. I can't see why there are permissions 'changed'. It is still a folder in /usr/share, there is nothing to hide here.

**Comment by bzed** [comment]
I've fixed the mentioned issues, but the tests still fail - somebody else has to look into it.

**Comment by mmoll** [comment]
try this:
```diff
--- a/spec/defines/collectd_plugin_python_spec.rb
+++ b/spec/defines/collectd_plugin_python_spec.rb
@@ -17,6 +17,7 @@ describe 'collectd::plugin::python', :type => :define do
         :content => "<LoadPlugin \"python\">\n    Globals true\n</LoadPlugin>\n\n\n",
       })
     end
+  end
 
   context ':ensure => absent' do
     let :facts do
```

File: manifests/plugin/python.pp [hunk 4]
```diff
@@ -20,26 +15,39 @@
...
-      mode    => '0640',
-      source  => $script_source,
-      require => File["${name}.load"],
-      notify  => Service['collectd'],
+      require => Package[$collectd::params::package]
+  }
+
+  collectd::plugin { 'python' :
+      ensure   => $ensure,
+      interval => $interval,
```
Comment (simonpasquier) [inline manifests/plugin/python.pp]: I'm pretty sure that the interval setting has no effect on the Python plugin.

Comment (txaj) [inline manifests/plugin/python.pp]: From the LoadPlugin doc : "[interval] Sets a plugin-specific interval for collecting metrics. This overrides the global Interval setting. If a plugin provides own support for specifying an interval, that setting will take precedence."

I am not familiar with Python nor the Python plugin of collectd ; http://collectd.org/documentation/manpages/collectd-python.5.shtml Are the notion of interval in the python plugin mandatory or could they inherit the value from collectd ?

Comment (bzed) [inline manifests/plugin/python.pp]: By default its inherited from collectd, but it is pretty fine to specify a completely different interval. And depending on what you do you in your python module you might want completely different intervals from the default.

Comment (simonpasquier) [inline manifests/plugin/python.pp]: Except that you missed this part: "If a plugin provides own support for specifying an interval, that setting will take precedence.". In the case of the Python plugin, there is no support for custom intervals so the parameter is ineffective. Here is what I get on one of my boxes:

    python plugin: Ignoring unknown config key "Interval"

**Comment by blkperl** [comment]
Rebase your changes to get the fixed tests

**Comment by blkperl** [comment]
So #255 was submitted as well. Please work together to figure out which PR should be merged.

**Comment by blkperl** [comment]
Merged #255 which resolves this.

**Pull request closed without merging**